File: ruby/palindrome.rb
# is_palindrome

def is_palindrome(string)
  reversed_string = []
  index = string.length - 1

  while (index >= 0)
    reversed_string << string[index]
    index -= 1
  end
  string == reversed_string.join()
end

p is_palindrome("racecar")

File: javascript/String/palindrome/palindrome.js

function isPalindrome(string) {
  let reversedString = []
  for (let i = string.length - 1; i >= 0; i--) {
    reversedString.push(string[i])
  }
  return string === reversedString.join("")
}

console.log(isPalindrome("racecar"))